streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -104,6 +104,18 @@
     private final Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> standbyRecords;
     private boolean processStandbyRecords = false;
 
+    static File makeStateDir(String jobId, String baseDirName) {
```

Comment:
Does `FIle.mkdirs` not work here? Seems like we could avoid this utility entirely.